zcash_client_sqlite/src/lib.rs

```diff
@@ -224,6 +225,39 @@ impl<P: consensus::Parameters + Clone> WalletDb<Connection, P> {
     }
 }
 
+impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletDb<C, P> {
+    pub fn read_transactionally<F, A, E: From<rusqlite::Error>>(&self, f: F) -> Result<A, E>
```

Comment:
TODO: do we also need `transactionally` to retry? If we only had `transactionally` then we would not, but I think it is possible for a call to `transaction()` to fail with `ErrorCode::DatabaseBusy` when there is an outstanding transaction opened by `unchecked_transaction` (this is not prevented by the type system).